ActivityLauncherApp/src/main/res/layout/dialog_edit_activity.xml

```diff
@@ -82,6 +82,62 @@
                 android:inputType="text" />
         </TableRow>
 
+
+        <TableRow>
+            <TextView
+                android:id="@+id/textView_action"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_gravity="center_vertical"
+                android:labelFor="@+id/editText_action"
+                android:text="@string/label_action"
+                android:textAppearance="?android:attr/textAppearanceMedium" />
+
+            <EditText
+                android:id="@+id/editText_action"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_weight="1"
+                android:importantForAutofill="no"
+                android:inputType="text"
+                android:text="android.intent.action.VIEW" />
+
+            <CheckBox
```

Comment:
Instead of using this checkbox, make the EditText into a selection or combo box. Also avoids the action to be nullable. We could hide the two new options also under an expansion panel for "advanced settings".